src/main/tray/tray.ts

```diff
@@ -16,151 +16,155 @@ import SettingsWindow from 'main/windows/settingsWindow';
+
+        switch (process.platform) {
+        case 'win32':
+            this.images = {
+                normal: nativeImage.createFromPath(path.resolve(assetsDir, `windows/tray_${winTheme}.ico`)),
+                unread: nativeImage.createFromPath(path.resolve(assetsDir, `windows/tray_${winTheme}_unread.ico`)),
+                mention: nativeImage.createFromPath(path.resolve(assetsDir, `windows/tray_${winTheme}_mention.ico`)),
             };
-        } else {
-            //Fallback for invalid theme setting
-            trayImages = {
-                normal: nativeImage.createFromPath(path.resolve(assetsDir, 'linux', 'top_bar_light_16.png')),
-                unread: nativeImage.createFromPath(path.resolve(assetsDir, 'linux', 'top_bar_light_unread_16.png')),
-                mention: nativeImage.createFromPath(path.resolve(assetsDir, 'linux', 'top_bar_light_mention_16.png')),
+            break;
+        case 'darwin':
+        {
+            const osxNormal = nativeImage.createFromPath(path.resolve(assetsDir, 'osx/menuIcons/MenuIcon16Template.png'));
+            const osxUnread = nativeImage.createFromPath(path.resolve(assetsDir, 'osx/menuIcons/MenuIconUnread16Template.png'));
+            osxNormal.setTemplateImage(true);
+            osxUnread.setTemplateImage(true);
+
+            this.images = {
+                normal: osxNormal,
+                unread: osxUnread,
+                mention: osxUnread,
             };
+
+            break;
         }
-        break;
-    }
-    default:
-        trayImages = {};
+        case 'linux':
+        {
+            if (trayIconTheme === 'dark') {
+                this.images = {
+                    normal: nativeImage.createFromPath(path.resolve(assetsDir, 'linux', 'top_bar_dark_16.png')),
+                    unread: nativeImage.createFromPath(path.resolve(assetsDir, 'linux', 'top_bar_dark_unread_16.png')),
+                    mention: nativeImage.createFromPath(path.resolve(assetsDir, 'linux', 'top_bar_dark_mention_16.png')),
+                };
+            } else {
+                //Fallback for invalid theme setting
+                this.images = {
+                    normal: nativeImage.createFromPath(path.resolve(assetsDir, 'linux', 'top_bar_light_16.png')),
+                    unread: nativeImage.createFromPath(path.resolve(assetsDir, 'linux', 'top_bar_light_unread_16.png')),
+                    mention: nativeImage.createFromPath(path.resolve(assetsDir, 'linux', 'top_bar_light_mention_16.png')),
+                };
+            }
+            break;
+        }
+        default:
+            this.images = {};
+        }
+        if (this.tray) {
+            this.update(this.status, this.message);
+        }
+        return this.images;
     }
-    if (trayIcon) {
-        setTray(lastStatus, lastMessage);
+
+    destroy = () => {
+        if (process.platform === 'win32') {
+            this.tray?.destroy();
+        }
```

Comment:
Well, this seems to come from a very very old change: https://github.com/mattermost/desktop/commit/932f7565706651d7af58ec741c265f9778370895#diff-27653c212e1cfe533e4eb2f7d0d3f89604c9de48a09583b4cbbbcbd08a07da79R168

So it's been around a while. Personally I'd like to keep it as is if it's not causing any issues, but would be interesting to see if this can be changed.